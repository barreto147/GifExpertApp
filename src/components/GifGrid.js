//import { useState, useEffect } from "react";
//import { getGifs } from "../helpers/getGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      <div className="card-grid animate__animated animate__zoomInUp ">
        {loading && <p>Loading</p>}
        {images.map((image) => {
          return <GifGridItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
