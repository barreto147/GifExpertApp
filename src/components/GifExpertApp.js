import { useState } from "react";
import { CategoryAdd } from "./CategoryAdd";
import { GifGrid } from "./GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  //   const handleAdd = () => {
  //     //setCategories([...categories, "Digimon"]);
  //     setCategories((categ) => [...categ, "Digimon"]);
  //   };

  return (
    <>
      <h2>GifExpertApp</h2>
      <CategoryAdd setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
