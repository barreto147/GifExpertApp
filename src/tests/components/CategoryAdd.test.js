import { CategoryAdd } from "../../components/CategoryAdd";
import { shallow } from "enzyme";

describe("Pruebas de CategoryAdd", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<CategoryAdd setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoryAdd setCategories={setCategories} />);
  });

  test("Prueba de Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";

    input.simulate("change", {
      target: {
        value,
      },
    });
    const inputAfter = wrapper.find("input");
    expect(inputAfter.prop("value")).toBe(value);
  });

  test("no debe de postear la informacion con submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("debe de llamar setCategories y limpiar la caja de texto", () => {
    const value = "Hola galaxia";

    wrapper.find("input").simulate("change", {
      target: {
        value,
      },
    });

    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    const inputAfter = wrapper.find("input");
    expect(inputAfter.prop("value")).toBe("");
  });
});
