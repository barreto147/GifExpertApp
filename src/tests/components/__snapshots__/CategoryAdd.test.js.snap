// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de CategoryAdd Prueba de Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
